packages/lesswrong/lib/analyticsEvents.tsx

```diff
@@ -404,11 +410,13 @@ function serverConsoleLogAnalyticsEvent(event: any) {
 
 let pendingAnalyticsEvents: Array<any> = [];
 
-export function flushClientEvents() {
+export function flushClientEvents(force: boolean = false) {
   if (!isClient)
     throw new Error("This function can only be run on the client");
   if (!pendingAnalyticsEvents.length)
     return;
+  if (!force && !clientContextVars.tabId)
+    return;
```

Comment:
What's the purpose of this?

Reply:
This is part of the fix for this bug:
> Fixes a bug in the 'pageLoadFinished' event, which caused it to be missing clientId, userId etc in some cases. This is the only event that sends utm params so it will help a lot to have them included here. An attempt was made to fix this bug before but it appears it didn't work

Basically in some cases this was called before the clientContextVars are set (which happens after the first render), so previously the pageLoadFinished event would be sent without the context vars (which means we can't associate the utm params with a user/client)